packages/web-components/fast-foundation/src/search/search.ts

```diff
@@ -0,0 +1,244 @@
+import { attr, DOM, nullableNumberConverter, observable } from "@microsoft/fast-element";
+import {
+    ARIAGlobalStatesAndProperties,
+    StartEnd,
+    StartEndOptions,
+} from "../patterns/index";
+import { applyMixins } from "../utilities/index";
+import type { FoundationElementDefinition } from "../foundation-element";
+import { FormAssociatedSearch } from "./search.form-associated";
+/**
+ * Search configuration options
+ * @public
+ */
+export type SearchOptions = FoundationElementDefinition & StartEndOptions;
+
+/**
+ * A Search Custom HTML Element.
+ * Based largely on the {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search | <input type="search" /> element }.
+ *
+ * @public
+ */
+export class Search extends FormAssociatedSearch {
+    /**
+     * When true, the control will be immutable by user interaction. See {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly | readonly HTML attribute} for more information.
+     * @public
+     * @remarks
+     * HTML Attribute: readonly
+     */
+    @attr({ attribute: "readonly", mode: "boolean" })
+    public readOnly: boolean;
+    private readOnlyChanged(): void {
+        if (this.proxy instanceof HTMLInputElement) {
+            this.proxy.readOnly = this.readOnly;
+            this.validate();
+        }
```

Comment:
Maybe all of this duplicate code could be consolidated into a method that does this.

```
private updateProxy(proxyAttr: string, searchAttr?: string) {
    if (this.proxy instanceof HTMLInputElement) {
        this.proxy[proxyAttr] = this[searchAttr || proxyAttr];
        this.validate();
    }
}
```

Then you could just call `this.updateProxy('readOnly')`

Reply:
this would work for everything but `list` since its readOnly